src/Orchard.Web/Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -60,55 +57,54 @@ public class AutoroutePartHandler : ContentHandler {
         private void PublishAlias(AutoroutePart part) {
             ProcessAlias(part);
 
-            // should it become the home page ?
-            if (part.DisplayAlias == "/") {
-                part.DisplayAlias = String.Empty;
-
-                // regenerate the alias for the previous home page
-                var currentHomePages = _orchardServices.ContentManager.Query<AutoroutePart, AutoroutePartRecord>().Where(x => x.DisplayAlias == "").List();
-                foreach (var current in currentHomePages.Where(x => x.Id != part.Id)) {
-                    if (current != null) {
-                        current.CustomPattern = String.Empty; // force the regeneration
-                        current.DisplayAlias = _autorouteService.Value.GenerateAlias(current);
-
-                        // we changed the alias of the previous homepage, so publish this change if the content item was published.
-                        if(current.IsPublished())
-                            _orchardServices.ContentManager.Publish(current.ContentItem);
+            // Should it become the home page?
+            if (part.PromoteToHomePage) {
+                // Get the current homepage an unmark it as the homepage.
+                var currentHomePage = _homeAliasService.GetHomePage(VersionOptions.Latest);
+                if(currentHomePage != null && currentHomePage.Id != part.Id) {
+                    var autoroutePart = currentHomePage.As<AutoroutePart>();
+
+                    if (autoroutePart != null) {
+                        autoroutePart.PromoteToHomePage = false;

```

Comment:
We do need to reset the precious homepage's `PromoteToHomePage` to `false`, because it is set to `true` by the driver when the user checks the "make this the homepage" checkbox.